pkg/util/net.go

```diff
@@ -142,3 +146,37 @@ func addNoProxyEnvVarEntries(entries map[string]struct{}, envVar string) {
 		}
 	}
 }
+
+func selectV4IPFromHostInterface() (string, error) {
+	ifaces, err := tcpnet.Interfaces()
+	if err != nil {
+		return "", err
+	}
+
+	// get list of interfaces
+	for _, i := range ifaces {
+		addrs, err := i.Addrs()
+		if err != nil {
+			klog.Warningf("failed to get IPs for interface %s: %v", i.Name, err)
+			continue
+		}
+
+		for _, addr := range addrs {
+			var ip tcpnet.IP
+			switch v := addr.(type) {
+			case *tcpnet.IPAddr:
+				// ignore IPv6
```

Comment:
This is OK for now, but at some point we won't be able to ignore IPv6.

Reply:
We have a separate epic for supporting IPv6, so we can handle it then.